### PR TITLE
[fix](cloud) Fix replication_num properties when create table and MTMV

### PR DESCRIPTION
**Pull request by gavinchou**
The properties rewrite policy is not applied for CreateTableStmt and CreateMTMVStmt because the `analyze()` of CreateTableStmt is not called when executed by nereids.

There are some properties, e.g. replication_num, are not correctly processed in cloud mode, which will cause the following error

```
2024-06-09 02:30:42,602 WARN (mtmv-task-execute-1-thread-1|139) [OlapInsertExecutor.onFail():218] insert [label_87cd2cb386774f4a_a14a00e9c7d93a51] with query id 87cd2cb386774f4a-a14a00e9c7d93a51 failed
org.apache.doris.nereids.exceptions.AnalysisException: errCode = 3, detailMessage = tablet 469525 alive replica num 1 < load required replica num 2, alive backends: [10002], detail: Visible Replicas:Visible version: 1, Replicas: [replicaId=469526, backendId=-1, backend=null, version=1, state=NORMAL]. or you may not have permission to access the current cluster = gavin_cluster_name0
        at org.apache.doris.nereids.trees.plans.commands.insert.OlapInsertExecutor.finalizeSink(OlapInsertExecutor.java:152) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertIntoTableCommand.initPlan(InsertIntoTableCommand.java:200) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertIntoTableCommand.runInternal(InsertIntoTableCommand.java:219) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertIntoTableCommand.run(InsertIntoTableCommand.java:103) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertOverwriteTableCommand.runInsertCommand(InsertOverwriteTableCommand.java:185) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertOverwriteTableCommand.insertInto(InsertOverwriteTableCommand.java:235) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.InsertOverwriteTableCommand.run(InsertOverwriteTableCommand.java:164) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.extensions.mtmv.MTMVTask.exec(MTMVTask.java:235) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.extensions.mtmv.MTMVTask.run(MTMVTask.java:200) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.task.AbstractTask.runTask(AbstractTask.java:167) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.extensions.mtmv.MTMVTask.runTask(MTMVTask.java:310) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.executor.DefaultTaskExecutorHandler.onEvent(DefaultTaskExecutorHandler.java:50) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.job.executor.DefaultTaskExecutorHandler.onEvent(DefaultTaskExecutorHandler.java:33) ~[doris-fe.jar:1.2-SNAPSHOT]
        at com.lmax.disruptor.WorkProcessor.run(WorkProcessor.java:143) ~[disruptor-3.4.4.jar:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: org.apache.doris.common.UserException: errCode = 3, detailMessage = tablet 469525 alive replica num 1 < load required replica num 2, alive backends: [10002], detail: Visible Replicas:Visible version: 1, Replicas: [replicaId=469526, backendId=-1, backend=null, version=1, state=NORMAL]. or you may not have permission to access the current cluster = gavin_cluster_name0
        at org.apache.doris.planner.OlapTableSink.createLocation(OlapTableSink.java:642) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.planner.OlapTableSink.complete(OlapTableSink.java:226) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.nereids.trees.plans.commands.insert.OlapInsertExecutor.finalizeSink(OlapInsertExecutor.java:126) ~[doris-fe.jar:1.2-SNAPSHOT]
        ... 14 more
```

### CODE REVIEW

**Comment by gavinchou** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by gavinchou** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java [hunk 1]
```diff
@@ -189,6 +189,7 @@ public CreateTableStmt(boolean ifNotExists, boolean isExternal, TableName tableN
         this.partitionDesc = partitionDesc;
         this.distributionDesc = distributionDesc;
         this.properties = properties;
+        PropertyAnalyzer.getInstance().rewriteForceProperties(this.properties);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:192]: better to rewrite them in Nereids CreateTableCommand

**Comment by gavinchou** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java (hunk 1 above)
Comment (gavinchou) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:193]: Try to rewrite properties in all possible path.

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by morrySnow — DISMISSED** [review]

**Comment by gavinchou** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**